### PR TITLE
chore(scripts): enhance deploy with added bandwidth checking

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Verify available Netlify bandwidth prior to staging deploy.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "jsx": "preserve",
     "sourceMap": true,
     "strict": true,
+    "skipLibCheck": true,
```
Comment (jzempel) [inline tsconfig.json:8]: Definitely less concerned about this here than on https://github.com/zendeskgarden/react-components/pull/1175.

File: utils/deploy.js [hunk 2]
```diff
@@ -22,20 +22,30 @@ envalid.cleanEnv(process.env, {
...
+        const result = await garden.netlifyDeploy({
+          dir,
+          production,
+          message
+        });
+
+        return result;
+      };
+
+      url = await garden.githubDeploy({ command, production });
```
Comment (hzhu) [inline utils/deploy.js:43]: Out of curiosity, if the GitHub deploy failed, would the error be bubbled up somewhere and handled?

**Review by hzhu — APPROVED** [review]

File: utils/deploy.js (hunk 2 above)
Comment (jzempel) [reply utils/deploy.js:43]: Yes. Note the caught, logged error below along with `process.exitCode = 1;` which would result in a CI fail.

Comment (hzhu) [reply utils/deploy.js:43]: Thanks 👍🏽

**Pull request merged**